wire-schema/src/main/java/com/squareup/wire/schema/ProtoFile.java

```diff
@@ -13,64 +13,40 @@
  */
-package com.squareup.wire.schema;
-
-import com.google.common.collect.ImmutableList;
-import com.squareup.wire.schema.internal.parser.ProtoFileElement;
-import java.util.List;
-import java.util.Set;
-
-import static com.squareup.wire.schema.Options.FILE_OPTIONS;
-
-public final class ProtoFile {
-  static final ProtoMember JAVA_PACKAGE = ProtoMember.get(FILE_OPTIONS, "java_package");
-
-  private final Location location;
-  private final List<String> imports;
-  private final List<String> publicImports;
-  private final String packageName;
-  private final List<Type> types;
-  private final List<Service> services;
-  private final List<Extend> extendList;
-  private final Options options;
-  private final Syntax syntax;
-  private Object javaPackage;
-
-  private ProtoFile(Location location, List<String> imports,
-      List<String> publicImports, String packageName, List<Type> types,
-      List<Service> services, List<Extend> extendList, Options options,
-      Syntax syntax) {
-    this.location = location;
-    this.imports = imports;
-    this.publicImports = publicImports;
-    this.packageName = packageName;
-    this.types = types;
-    this.services = services;
-    this.extendList = extendList;
-    this.options = options;
-    this.syntax = syntax;
-  }
-
-  static ProtoFile get(ProtoFileElement protoFileElement) {
-    String packageName = protoFileElement.getPackageName();
-
-    ImmutableList<Type> types = Type.fromElements(packageName, protoFileElement.getTypes());
-
-    ImmutableList<Service> services =
-        Service.fromElements(packageName, protoFileElement.getServices());
-
-    ImmutableList<Extend> wireExtends =
-        Extend.fromElements(packageName, protoFileElement.getExtendDeclarations());
-
-    Options options = new Options(Options.FILE_OPTIONS, protoFileElement.getOptions());
-
-    return new ProtoFile(protoFileElement.getLocation(), protoFileElement.getImports(),
-        protoFileElement.getPublicImports(), packageName, types, services, wireExtends, options,
-        protoFileElement.getSyntax());
-  }
-
-  ProtoFileElement toElement() {
-    return new ProtoFileElement(
+package com.squareup.wire.schema
+
+import com.squareup.wire.schema.Extend.Companion.fromElements
+import com.squareup.wire.schema.ProtoMember.Companion.get
+import com.squareup.wire.schema.Service.Companion.fromElements
+import com.squareup.wire.schema.Type.Companion.fromElements
+import com.squareup.wire.schema.internal.parser.ProtoFileElement
+
+class ProtoFile private constructor(
+  private val location: Location,
+  private val imports: List<String>,
+  private val publicImports: List<String>,
+  private val packageName: String?,
+  private val types: List<Type>,
+  private val services: List<Service>,
+  private val extendList: List<Extend>,
+  private val options: Options,
+  private val syntax: Syntax?
+) {
+  private var javaPackage: Any? = null
+
+  // TODO(Benoit) Set all `val` as public and remove the redundant getters.
```

Comment:
what worked for me was renaming the accessors to use a `get` prefix, _then_ doing the Convert to Kotlin shortcut

Reply:
Yes, I wanna do that. It's just that more than 15 files get updated. I thought that deserved its own PR